### PR TITLE
fix: handle None stdio streams in macOS child process bootstrap

### PR DESCRIPTION
**Pull request by ajcasagrande**
On macOS, FD_CLOEXEC is set on terminal FDs before spawning child processes. After exec, FDs 0/1/2 are closed and Python sets sys.stdout/stderr to None. The old monolithic try/except silently swallowed the AttributeError on None.close(), skipping the devnull reassignment entirely. Libraries like billiard then crash on sys.stdout.flush() during ProcessPoolExecutor spawning.

Fix by opening devnull files unconditionally first, then closing old streams individually (tolerating None), then always assigning.

## Summary by CodeRabbit

* **Refactor**
  * Centralized STDIO redirection for macOS child processes to improve reliability, error propagation, and handling of edge cases where streams may be missing.

* **Tests**
  * Added macOS-specific tests validating STDIO redirection across different stream states (none, real files, mixed) and ensuring proper restoration and cleanup.

### CODE REVIEW

File: src/aiperf/common/bootstrap.py [hunk 1]
```diff
@@ -172,6 +159,28 @@ async def _run_service():
...
+    """Redirect stdin/stdout/stderr to /dev/null for macOS child processes.
+
+    Prevents child processes from accessing the parent's terminal, which causes
+    Textual UI corruption (ASCII garbage and freezes from inherited terminal FDs).
+    Handles the case where streams are already None (e.g., in spawned contexts)
+    to avoid AttributeError when libraries like billiard call sys.stdout.flush().
+    """
+    devnull_stdin = open(os.devnull)  # noqa: SIM115
+    devnull_stdout = open(os.devnull, "w")  # noqa: SIM115
+    devnull_stderr = open(os.devnull, "w")  # noqa: SIM115
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply src/aiperf/common/bootstrap.py]: fixed

File: tests/unit/common/test_bootstrap_macos.py [hunk 2]
```diff
@@ -136,3 +141,61 @@ def test_terminal_fd_closing_handles_exceptions(
...
+                sys.stdin = open(os.devnull)  # noqa: SIM115
+                sys.stdout = open(os.devnull, "w")  # noqa: SIM115
+                sys.stderr = open(os.devnull, "w")  # noqa: SIM115
+
+                _redirect_stdio_to_devnull()
+
+                assert sys.stdout.name == os.devnull
+                assert sys.stderr.name == os.devnull
+                assert sys.stdin.name == os.devnull
+            finally:
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply tests/unit/common/test_bootstrap_macos.py]: fixed
